lib/clamp-link.in

```diff
@@ -523,8 +523,9 @@ do
 
       FILE=`basename "$OBJ"` # remove path
       FILENAME="${FILE%.*}"
-      KERNEL_FILE="$TEMP_DIR/$FILENAME.kernel.bc"
-      HOST_FILE="$TEMP_DIR/$FILENAME.host.o"
+      HASH=($(md5sum $OBJ))
```

Comment:
I started off with Random numbers in bash for this purpose, but Sameer and Damon  cDougall, pointed out that predictable names are better for build process and debugging with several runs. So I changed the code to use md5sum. 
Further md5sum is also a part of coreutils. So I believe this may not add extra package dependency. (?)

Reply:
I agree on having more predictable names and you correct that md5sum in coreutils